File: TeamCode/src/main/java/com/info1robotics/bobot/tasks/InlineTask.java
package com.info1robotics.bobot.tasks;

/**
 * Runs the given block synchronously.
 * Intended for one-liners.
 */
public class InlineTask extends Task {
    private final Runnable runnable;
    public InlineTask(Runnable runnable) {
        this.runnable = runnable;
    }

    @Override
    public void run() {
        runnable.run();
        state = State.FINISHED;
    }
}

File: TeamCode/src/main/kotlin/com/info1robotics/bobot/tasks/InlineTask.kt
package com.info1robotics.bobot.tasks

/**
 * Wraps the given block in a task.
 * Intended for one-liners.
 * @param task Lambda to be executed.
 */
class InlineTask(private val task: () -> Any) : Task() {
    override fun run() {
        task()
        state = State.FINISHED
    }
}